konveyor-base/src/main/java/io/github/magwas/runtime/LogUtil.java

```diff
@@ -30,18 +30,11 @@ public static void debug(final int depth, final Object... args) {
 		StackTraceElement stackTraceElement = Thread.currentThread().getStackTrace()[depth];
 		String name = stackTraceElement.getClassName();
 		if (!debuggedClasses.contains(name)) return;
-		List<String> params = List.of(args).stream().map(Object::toString).collect(Collectors.toList());
+		List<String> params = Stream.of(args).map(Object::toString).toList();
 		String method = stackTraceElement.getMethodName();
-		StringBuilder builder = new StringBuilder();
-		builder.append("DEBUG ");
-		builder.append(name);
-		builder.append(" ");
-		builder.append(method);
-		builder.append(" ");
-		builder.append(stackTraceElement.getLineNumber());
-		builder.append(":");
-		builder.append(String.join(",", params));
-		System.err.println(builder.toString());
+		String string = "DEBUG " + name + ' ' + method + ' ' + stackTraceElement.getLineNumber() + ':'
```

Comment:
I need some convincing here. StringBuilder might be too resource-intensive, but string concatenation of so many elements is hard to read. There are plenty of alternatives, including MessageFormat. Let's discuss what is best for this use case, and what rule should we adapt.